source/base/function_signed_distance.cc

```diff
@@ -353,6 +353,166 @@ namespace Functions
 
       return bounding_box.signed_distance(p);
     }
+
+
+
+    template <int dim>
+    ZalesakDisk<dim>::ZalesakDisk(const Point<dim> &center,
+                                  const double      radius,
+                                  const double      notch_width,
+                                  const double      notch_height)
+      : sphere(center, radius)
+      , notch(
+          {// bottom left
+           (dim == 1) ?
+             Point<dim>(center[0] - 0.5 * notch_width) :
+           (dim == 2) ?
+             Point<dim>(center[0] - 0.5 * notch_width,
+                        center[1] - std::sqrt(std::pow(radius, 2) -
+                                              std::pow(0.5 * notch_width, 2))) :
+             Point<dim>(center[0] - 0.5 * notch_width,
+                        std::numeric_limits<
+                          double>::lowest() /* notch is open in y-direction*/,
```

Comment:
Nice job -- very few people know about the difference between `std::numeric_limits::lowest()` and `std::numeric_limit::min()` and either incorrectly use the latter, or give up and use `-std::numeric_limits::max()` :-)